src/dmd/dcast.d

```diff
@@ -3827,3 +3835,58 @@ IntRange getIntRange(Expression e)
     e.accept(v);
     return v.range;
 }
+
+Type checkArrayOpType(Expression e1, Expression e2, TOK op, Scope *sc)
+{
+    Expression lhs = e1;
```

Comment:
why not name the parameters lhs and rhs? That way, these declarations become useless.